packages/amplify-category-api/src/graphql-transformer/transformer-options-types.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * ProjectOptions Type Definition
+ */
+import {
+  AppSyncAuthConfiguration,
+  TransformerPluginProvider,
+} from '@aws-amplify/graphql-transformer-interfaces';
+import {
+  OverrideConfig,
+  ResolverConfig,
+  TransformerProjectConfig,
+} from '@aws-amplify/graphql-transformer-core';
+import Template from '@aws-amplify/graphql-transformer-core/lib/transformation/types';
+import {
+  DiffRule,
+  ProjectRule,
+} from 'graphql-transformer-core';
+import { $TSAny } from 'amplify-cli-core';
+
+/**
+ * Transformer Options used to create a GraphQL Transform and compile a GQL API
+ */
+export type TransformerProjectOptions<T> = {
+  buildParameters: {
+    S3DeploymentBucket: string;
+    S3DeploymentRootKey: string;
+  };
+  projectDirectory: string;
+  transformersFactory: (options: T) => Promise<TransformerPluginProvider[]>;
+  transformersFactoryArgs: T;
+  rootStackFileName: 'cloudformation-template.json';
+  currentCloudBackendDirectory?: string;
+  minify: boolean;
+  lastDeployedProjectConfig?: TransformerProjectConfig;
+  projectConfig: TransformerProjectConfig;
+  resolverConfig?: ResolverConfig;
+  dryRun?: boolean;
+  authConfig?: AppSyncAuthConfiguration;
+  stacks: Record<string, Template>;
+  sandboxModeEnabled?: boolean;
+  sanityCheckRules: SanityCheckRules;
+  overrideConfig: OverrideConfig;
+};
+
+type SanityCheckRules = {
+  diffRules: DiffRule[];
+  projectRules: ProjectRule[];
+};
+
+/**
+ * Arguments passed into a TransformerFactory
+ * Used to determine how to create a new GraphQLTransform
+ */
+export type TransformerFactoryArgs = {
+  addSearchableTransformer: boolean;
+  authConfig: $TSAny;
+  storageConfig?: $TSAny;
+  adminRoles?: Array<string>;
+  identityPoolId?: string;
```

Comment:
is this not part of the `authConfig`?

Reply:
unsure, I only moved some existing code into new files as I was trying to make it more manageable. This isn't a change